skbio/tree/tests/test_tree.py

```diff
@@ -1303,6 +1303,42 @@ def test_shuffle_raises(self):
         with self.assertRaises(MissingNodeError):
             next(self.simple_t.shuffle(names=['x', 'y']))
 
+    def test_cache_leafcount(self):
+        t = TreeNode.read([r'((a,b)e,(c,d)f,g)r;'])
+        t._cache_leafcount()
+        self.assertEquals(t.find('r').leafcount, 5)
```

Comment:
I'm concerned about this. The developer will need to know to issue a call to a private member method of the tree in order to have access to this public attribute. This present approach will lead to a lot of `AttributeErrors` from users. I think it would be more appropriate for these to be properties so that the caching and construction of the attributes lazy and seamless.